src/pybind/mgr/dashboard/controllers/docs.py

```diff
@@ -0,0 +1,264 @@
+            return "integer"
+        elif "count" in param_name:
+            return "integer"
+        elif "num" in param_name:
+            return "integer"
+        elif isinstance(def_value, bool):
+            return "boolean"
+        elif isinstance(def_value, int):
+            return "integer"
+        return "string"
+
+    @classmethod
+    def _gen_body_param(cls, body_params):
+        required = [p['name'] for p in body_params if p['required']]
+
+        props = {}
+        for p in body_params:
+            props[p['name']] = {
+                'type': cls._gen_type(p)
+            }
+            if 'default' in p:
+                props[p['name']]['default'] = p['default']
+
+        if not props:
+            return None
+
+        return {
+            'in': "body",
+            'name': "body",
+            'description': "",
+            'required': True,
+            'schema': {
+                'type': "object",
+                'required': required,
+                'properties': props
+            }
+        }
+
+    @classmethod
+    def _gen_responses_descriptions(cls, method):
+        resp = {
+            '400': {
+                "description": "Operation exception. Please check the "
+                               "response body for details."
+            },
+            '401': {
+                "description": "Unauthenticated access. Please login first."
+            },
+            '403': {
+                "description": "Unauthorized access. Please check your "
+                               "permissions."
+            },
+            '500': {
+                "description": "Unexpected error. Please check the "
+                               "response body for the stack trace."
+            }
+        }
+        if method.lower() == 'get':
+            resp['200'] = {'description': "OK"}
+        if method.lower() == 'post':
+            resp['201'] = {'description': "Resource created."}
+        if method.lower() == 'put':
+            resp['200'] = {'description': "Resource updated."}
+        if method.lower() == 'delete':
+            resp['204'] = {'description': "Resource deleted."}
+        if method.lower() in ['post', 'put', 'delete']:
+            resp['202'] = {'description': "Operation is still executing."
+                                          " Please check the task queue."}
+
+        return resp
+
+    @classmethod
+    def _gen_param(cls, param, ptype):
+        res = {
+            'name': param['name'],
+            'in': ptype,
+            'type': cls._gen_type(param)
+        }
+        if param['required']:
+            res['required'] = True
+        elif param['default'] is None:
+            res['allowEmptyValue'] = True
+        else:
+            res['default'] = param['default']
+        return res
+
+    def _gen_spec(self, all_endpoints=False, baseUrl=""):
+        if all_endpoints:
+            baseUrl = ""
+        METH_ORDER = ['get', 'post', 'put', 'delete']
```

Comment:
Nit: spell out "METHOD" here, to avoid shady mental associations to drug dealings: `METHOD_ORDER`